### ISSUE TITLE
Package cuSPARSELt

### ISSUE THREAD

**Issue by SomeoneSerge** [NixOS/nixpkgs#270446]
## Issue description

Introduce `cudaPackages.cusparselt` based on https://developer.download.nvidia.com/compute/cusparselt/redist/redistrib_0.5.0.json

This is required e.g. to remove the dependency on the runfile-based `cudatoolkit` in `python3Packages.cupy`

CC @NixOS/cuda-maintainers

**Comment by CertainLach** [comment]
Build of cupy fails on master with
```
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}: -------- Configuring Module: cusparselt --------
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}: /build/tmpi_pdh6sl/a.cpp:1:10: fatal error: cusparseLt.h: No such file or directory
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}:     1 | #include <cusparseLt.h>
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}:       |          ^~~~~~~~~~~~~~
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}: compilation terminated.
 deploy{host=XXX}:build{drv="python3.11-cupy-13.0.0.drv"}: command '/nix/store/kvlhk0gpm2iz1asbw1xjac2ch0r8kyw9-gcc-wrapper-13.2.0/bin/g++' failed with exit code 1
```

**Comment by prusnak** [comment]
@sepiabrown Are you working on packaging cusparselt? Do you plan to send a PR?

**Comment by sepiabrown** [comment]
> [@sepiabrown](https://github.com/sepiabrown) Are you working on packaging cusparselt? Do you plan to send a PR?

Yes, I'd like to. I'm trying to build `cusparselt` for `cuda-library-samples` but haven't succeeded.

**Comment by prusnak** [comment]
> I'm trying to build `cusparselt` for `cuda-library-samples` but haven't succeeded.

How do you build cuda-library-samples for cusparselt? I am having hard time to figure this one out.

**Comment by sepiabrown** [comment]
> > I'm trying to build `cusparselt` for `cuda-library-samples` but haven't succeeded.
> 
> How do you build cuda-library-samples for cusparselt? I am having hard time to figure this one out.

I have succeeded in building cuda-library-samples for cusparselt. There are some quirks that seem unnecessary. For example, I have built cusparselt based on cutensor and it doesn't populate the `lib64` folder but only populate `lib`. However, cuda-library-samples requires `lib64`, so I postPatched it like below.
```nix
+      postPatch = ''
+        substituteInPlace CMakeLists.txt \
+          --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt.so" "${lib.getLib cusparselt}/lib/libcusparseLt.so" \
+          --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt_static.a" "${lib.getStatic cusparselt}/lib/libcusparseLt_static.a"
+      '';
```

**Comment by prusnak** [comment]
Done in https://github.com/NixOS/nixpkgs/pull/390863
